src/main/java/org/datadog/jenkins/plugins/datadog/model/BuildPipeline.java

```diff
@@ -114,6 +114,12 @@ private void completeInformation(final List<BuildPipelineNode> nodes, final Buil
                 }
             }
 
+            // As the agent information will be stored in the Allocate node children
+            // we need to propagate the information to the indexed span.
+            for(final BuildPipelineNode parentIter : node.getParents()) {
+                parentIter.setPropagatedSecondsInQueue(node.getSecondsInQueue());
+            }
```

Comment:
I don't follow that bit. We're propagating the time in queue of the curent node to each of its parent and we only do that to make the retrieval of that information easier later on, right?

What if a parent has multiple children and they all update the `propagatedSecondsInQueue`? What about the time the parents themselves are in queue? There might be something I don't follow to understand fully this PR

Reply:
You're right, It can cause unexpected behaviour. I inverted the propagation approach, checking when the current node is the `initial` or `stage` and checking if its child is an `Allocate node` node. Only in this case, we propagate the queue time.

This is needed because in Jenkins the queue time is calculated in this kind of nodes (`Allocate node`), and not in the concrete `Stage`/ `Initial` node. We need to propagate all information to the correct nodes that will be indexed in the backend.